src/memory-analyzer/memory_analyzer_parse_options.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2018 Author: Malte Mues <mail.mues@gmail.com>
+
+/// \file
+/// Commandline parser for the memory analyzer executing main work.
+
+#ifdef __linux__
+
+#include "memory_analyzer_parse_options.h"
+#include "analyze_symbol.h"
+#include "gdb_api.h"
+
+#include <goto-programs/goto_model.h>
+#include <goto-programs/read_goto_binary.h>
+#include <util/config.h>
+#include <util/exit_codes.h>
+#include <util/message.h>
+#include <util/string_utils.h>
+
+int memory_analyzer_parse_optionst::doit()
+{
+  // This script is the entry point and has to make sure
+  // that the config object is set to a valid architecture.
+  // config is later used to determine right size for types.
+  // If config is not set, you might see a bunch of types with
+  // size 0.
+  // It might be desierable to convert this into flags in the future.
+  config.ansi_c.mode = configt::ansi_ct::flavourt::GCC;
+  config.ansi_c.set_arch_spec_x86_64();
```

Comment:
I don't think we should hardcode the architecture.  Is the usual option of having flags and defaulting to the platform architecture not viable?

Reply:
I found code that was supposed to read these settings from the goto-binary, but couldn't make that work with my goto binaries.
Anyhow, these information, if passed as a flag, should effect, how gdb is going to interpret the core dump.
Modifying these gdb settings isn't supported right now by the ```gdb_apit```.

So there is more work required than just changing these hard coded values into flags. To prevent users from passing unsupported flags not aligned with the gdb behavior, I hard coded it for now.

Reply:
I can see that it makes sense for now but it would be good to avoid this if we can.